### ISSUE TITLE
Support explicitly defining and parameterizing separate loss functions

### ISSUE THREAD

**Issue by phildow** [doc-ai/aleph#28]
Neeraj comments:
"Along with optimizers, we should also allow users to specify a loss function. Standard loss functions (cross entropy, mean squared error, etc.) or user-defined losses as we have used internally. Ideally, this would generate a $LOSS_NAME.py file which would get imported into the model file."
